vault/login_mfa.go

```diff
@@ -1848,50 +1850,34 @@ func (c *Core) validateOkta(ctx context.Context, mConfig *mfa.Config, username s
 		return fmt.Errorf("more than one user found for e-mail address")
 	}
 
-	user := &users[0]
+	user := users[0]
 
-	_, err = client.Users.PopulateMFAFactors(user)
+	factors, _, err := client.UserFactor.ListFactors(ctx, user.Id)
```

Comment:
This might be out of scope as well, but should we check for MFA enrollment status like we do on the Okta Auth Method? https://github.com/hashicorp/vault/blob/50afd00b0b92aad4282edcfeeea7d3aec8a64b0d/builtin/credential/okta/backend.go#L152